app/src/main/java/org/wikipedia/views/ImagePreviewDialog.kt

```diff
@@ -0,0 +1,143 @@
+import org.wikipedia.util.L10nUtil.setConditionalLayoutDirection
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.UriUtil
+import org.wikipedia.util.log.L
+
+
+class ImagePreviewDialog : ExtendedBottomSheetDialogFragment(), DialogInterface.OnDismissListener {
+
+
+    private var fileName: String? = null
+    private lateinit var imageInfo: ImageInfo
+
+    private val menuListener = PopupMenu.OnMenuItemClickListener { item ->
+        false
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val app = WikipediaApp.getInstance()
+
+        val rootView = inflater.inflate(R.layout.dialog_image_preview, container)
+
+        setConditionalLayoutDirection(rootView, WikipediaApp.getInstance().language().appLanguageCode)
+        imageInfo = arguments!!.getSerializable("imageInfo") as ImageInfo
+        fileName = arguments!!.getString("filename")
+        return rootView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        overflowButton!!.setOnClickListener { v: View ->
+            val popupMenu = PopupMenu(requireActivity(), overflowButton!!)
+            popupMenu.inflate(R.menu.menu_link_preview)
+            popupMenu.setOnMenuItemClickListener(menuListener)
+            popupMenu.show()
+        }
+
+        // show the progress bar while we load content...
+        progressBar!!.visibility = View.VISIBLE
+        toolbarView.setOnClickListener { dismiss() }
+        imagePageCommonsLinkContainer.setOnClickListener {
+            dismiss()
+            UriUtil.visitInExternalBrowser(context,
+                    Uri.parse(String.format(getString(R.string.suggested_edits_image_file_page_commons_link), fileName)))
+        }
+        setImageDetails()
+    }
+
+    override fun onDestroyView() {
+        toolbarView!!.setOnClickListener(null)
+        overflowButton!!.setOnClickListener(null)
+        super.onDestroyView()
+    }
+
+    private fun setImageDetails() {
+        loadImage(imageInfo.originalUrl)
+        titleText!!.text = StringUtil.fromHtml(fileName!!.removePrefix("File:"))
+        addDetailPortion(getString(R.string.suggested_edits_image_caption_title), StringUtil.fromHtml(imageInfo.getMetadata()!!.imageDescription()!!.value()).toString())
+        addDetailPortion(getString(R.string.suggested_edits_image_artist), StringUtil.fromHtml(imageInfo.getMetadata()!!.artist()!!.value()).toString())
+        addDetailPortion(getString(R.string.suggested_edits_image_date), imageInfo.getMetadata()!!.dateTime()!!.value())
+        addDetailPortion(getString(R.string.suggested_edits_image_source), imageInfo.getMetadata()!!.imageDescription()!!.source())
+        addDetailPortion(getString(R.string.suggested_edits_image_licensing), imageInfo.getMetadata()!!.licenseShortName()!!.value())
+        detailsHolder.requestLayout()
```

Comment:
Instead of using `imageInfo.getMetadata()`, you can directly use `imageInfo.metadata`

Reply:
Done